File: diffapp/test/opencv_ios_pbxproj_test.dart
import 'dart:io';

import 'package:flutter_test/flutter_test.dart';

void main() {
  test('iOS Xcode プロジェクトの pbxproj にネイティブ連携の下地がある', () async {
    const path = 'ios/Runner.xcodeproj/project.pbxproj';
    final file = File(path);
    expect(file.existsSync(), isTrue, reason: '$path が存在すること');

    final pbx = await file.readAsString();

    // Swift-ObjC ブリッジングヘッダ設定（ネイティブ連携でよく使う）
    expect(
      pbx.contains('SWIFT_OBJC_BRIDGING_HEADER = "Runner/Runner-Bridging-Header.h";'),
      isTrue,
      reason: 'SWIFT_OBJC_BRIDGING_HEADER が設定されていること',
    );

    // C++ ランタイム（libc++）の利用設定（OpenCV等のC++連携で前提となる）
    expect(
      pbx.contains('CLANG_CXX_LIBRARY = "libc++";'),
      isTrue,
      reason: 'CLANG_CXX_LIBRARY が libc++ に設定されていること',
    );

    // Frameworks ビルドフェーズが存在する（フレームワークリンクの前提）
    expect(
      pbx.contains('/* Frameworks */ = {') || pbx.contains('PBXFrameworksBuildPhase'),
      isTrue,
      reason: 'Frameworks のビルドフェーズが定義されていること',
    );

    // 実機向けデプロイターゲット（13以上）
    expect(
      RegExp(r'IPHONEOS_DEPLOYMENT_TARGET\s*=\s*1[3-9]\.').hasMatch(pbx),
      isTrue,
      reason: 'iOS のデプロイターゲットが 13 以上であること',
    );
  });
}

File: src/opencv_ios_pbxproj_test.ts
import { describe, it, expect } from 'vitest';
import { readFileSync } from 'node:fs';

describe('iOS: Xcode pbxproj のネイティブ連携下地', () => {
  const path = 'diffapp/ios/Runner.xcodeproj/project.pbxproj';
  const t = () => readFileSync(path, 'utf-8');

  it('Bridging Header が設定されている', () => {
    expect(t()).toMatch(/Runner-Bridging-Header\.h/);
    expect(t()).toMatch(/SWIFT_OBJC_BRIDGING_HEADER\s*=\s*"Runner\/Runner-Bridging-Header\.h"/);
  });

  it('libc++ が指定されている', () => {
    expect(t()).toMatch(/CLANG_CXX_LIBRARY\s*=\s*"libc\+\+"/);
  });

  it('Frameworks ビルドフェーズが存在する', () => {
    expect(t()).toMatch(/\/\* Frameworks \*\//);
    expect(t()).toMatch(/Embed Frameworks/);
  });

  it('iOS 13+ をターゲットとしている', () => {
    expect(t()).toMatch(/IPHONEOS_DEPLOYMENT_TARGET\s*=\s*13\.0/);
  });
});
